frontends/concrete-python/concrete/fhe/extensions/multivariate.py

```diff
@@ -0,0 +1,104 @@
+"""
+Declaration of `multivariate` extension.
+"""
+
+from copy import deepcopy
+from typing import Any, Callable, Optional, Type, Union
+
+import numpy as np
+
+from ..dtypes import BaseDataType, Float
+from ..representation import Node
+from ..tracing import ScalarAnnotation, Tracer
+from ..values import ValueDescription
+
+
+def multivariate(
+    function: Callable,
+    outputs: Optional[Union[BaseDataType, Type[ScalarAnnotation]]] = None,
+) -> Callable:
+    """
+    Wrap a multivariate function so that it is traced into a single generic node.
+
+    Args:
+        function (Callable[[Any, ...], Any]):
+            multivariate function to wrap
+
+        outputs (Optional[Union[BaseDataType, Type[ScalarAnnotation]]], default = None):
+            data type of the result, unused during compilation, required for direct definition
```

Comment:
`unused during compilation` and `direct definition` is unclear to me.
is `outputs a hint  ? what is direct definition ?

Reply:
https://docs.zama.ai/concrete/tutorials/direct_circuits

Reply:
It's the API in which there is no inputset and dtypes are manually assigned. It's pretty much impossible to know the bit-width of a multivariate operation without evaluating it and there is no evaluation in direct definition. So users need to provide the resulting bit-width.

```python
fhe.univariate(function, outputs=fhe.uint4)(x)
```